greynoise/icon_greynoise/connection/connection.py

```diff
@@ -0,0 +1,41 @@
+import insightconnect_plugin_runtime
+from .schema import ConnectionSchema
+
+# Custom imports below
+import re
+from greynoise import GreyNoise
+from greynoise.exceptions import RequestFailure
+from insightconnect_plugin_runtime.exceptions import ConnectionTestException
+
+PLUGIN_STRING = open("/python/src/plugin.spec.yaml", "r").read()
+PLUGIN_VERSION = re.search(r"\bversion:\s(.*)", PLUGIN_STRING).group(1)
```

Comment:
@mrinehart-r7 can probably point to a better way to do this

Reply:
@bradchiappetta the plugin version can be obtained through the following code snippet (with self referring to a connection object):
```
self.meta.version
```
Or from an action/trigger like so (since each action/trigger has a connection property):
```
self.connection.meta.version
```
You can also get the plugin name, vendor, and description in this same fashion.